src/datasets/builder.py

```diff
@@ -1039,6 +1044,11 @@ def as_streaming_dataset(
         )
         if isinstance(datasets, dict):
             datasets = IterableDatasetDict(datasets)
+            if split == "all":
+                datasets = _concatenate_iterable_datasets(
+                    datasets.values(), info=self.info, split="+".join(splits_generator), axis=0
+                )
```

Comment:
Nice thanks !

I think you can place this inside _as_streaming_dataset_single instead for consistency with as_dataset/_build_single_dataset